### PR TITLE
Fix trustchain resolution for HTTP and mobile

### LINKED ISSUE #153: Fix trustchain resolution for HTTP and mobile
- The `Resolver<T>` used within `trustchain-http` is currently wrapping another `Resolver<T>`, both implementing `TrustchainResolver`. As a result, `trustchain_resolve()` is called twice and verification methods from CAS keys are added multiple times. Two proposed revisions:
  - Revise the `Resolver<T>` in `trustchain-http` to wrap an ION `DIDResolver` not a `TrustchainResolver`
  - Include a check when adding verification methods that a given verification method is not already present
- Since including an `IPFSClient` within `Resolver<T>`, it will no longer be possible to call [`verify()`](https://github.com/alan-turing-institute/trustchain/blob/96d8809228e964dfebd882f8f1038de01d5c7774/trustchain-core/src/verifier.rs#L205) on mobile as the `IPFSClient` cannot be used. One solution is to use the type state pattern as for the `IONVerifier` and use a `trustchain-http` endpoint for resolution within the mobile. A more complete refactor removing the `Resolver<T>` completely from the API in favour of generics that `impl TrustchainResolver` but this may be better considered in more detail in a further issue.

### PR DESCRIPTION
**Pull request by sgreenbury**
Closes #153.

### CODE REVIEW

File: trustchain-ion/src/lib.rs [hunk 1]
```diff
@@ -52,9 +53,21 @@ impl Endpoint {
...
+}
+
+/// Trustchain ION DID resolver for full client.
+pub fn trustchain_resolver(endpoint: &str) -> Resolver<HTTPSidetreeDIDResolver<ION>> {
+    Resolver::<_, FullClient>::new(ion_resolver(endpoint))
+}
+
+/// Trustchain ION DID resolver for full client.
+pub fn trustchain_resolver_light_client(
+    endpoint: &str,
```
Comment (sgreenbury) [inline trustchain-ion/src/lib.rs:68]:
```suggestion
    trustchain_endpoint: &str,
```

File: trustchain-ion/src/lib.rs [hunk 2]
```diff
@@ -52,9 +53,21 @@ impl Endpoint {
     }
 }
 
-/// Test resolver
-pub fn get_ion_resolver(endpoint: &str) -> IONResolver {
-    IONResolver::from(SidetreeClient::<ION>::new(Some(String::from(endpoint))))
+/// ION DID resolver.
+pub fn ion_resolver(endpoint: &str) -> HTTPSidetreeDIDResolver<ION> {
```
Comment (sgreenbury) [inline trustchain-ion/src/lib.rs:57]:
```suggestion
pub fn http_resolver(endpoint: &str) -> HTTPSidetreeDIDResolver<ION> {
```

**Review by thobson88 — APPROVED** [review]
Awesome

**Review by edchapman88 — APPROVED** [review]
looks good!

**Pull request merged**